WalletWasabi/Tor/TorMonitor.cs

```diff
@@ -144,26 +146,45 @@ await foreach (TorControlReply reply in client.ReadEventsAsync(cancellationToken
 	/// <inheritdoc/>
 	protected override async Task ActionAsync(CancellationToken token)
 	{
-		if (TorHttpPool.TorDoesntWorkSince is { }) // If Tor misbehaves.
+		if (TorHttpPool.TorDoesntWorkSince is not null) // If Tor misbehaves.
 		{
 			TimeSpan torMisbehavedFor = DateTimeOffset.UtcNow - TorHttpPool.TorDoesntWorkSince ?? TimeSpan.Zero;
 
 			if (torMisbehavedFor > CheckIfRunningAfterTorMisbehavedFor)
 			{
 				if (TorHttpPool.LatestTorException is TorConnectCommandFailedException torEx)
 				{
-					if (torEx.RepField == RepField.HostUnreachable)
+					if (torEx.RepField == RepField.OnionServiceIntroFailed)
 					{
-						Logger.LogInfo("Tor does not work properly. Test fallback URI.");
-						using HttpRequestMessage request = new(HttpMethod.Get, FallbackBackendUri);
-						using HttpResponseMessage _ = await HttpClient.SendAsync(request, token).ConfigureAwait(false);
+						Logger.LogInfo("Tor cannot access remote host. Test fallback URI.");
 
-						// Check if it changed in the meantime...
-						if (TorHttpPool.LatestTorException is TorConnectCommandFailedException torEx2 && torEx2.RepField == RepField.HostUnreachable)
+						// Check if the fallback address (clearnet) works. It must work for us to switch to the fallback address.
+						try
+						{
+							using HttpRequestMessage request = new(HttpMethod.Get, FallbackBackendUri);
+							using HttpResponseMessage _ = await HttpClient.SendAsync(request, token).ConfigureAwait(false);
+						}
+						catch
+						{
+							// The fallback address does not work too. We do not want to switch.
+							return;
+						}
+
+						// Check if the onion address is still broken.
+						try
+						{
+							using HttpRequestMessage request = new(HttpMethod.Get, BackendUri);
```

Comment:
We got here because Tor was not working, don't need to check it again here. It just adds another unexpected condition that can fail and prevent fallback.